flocker/node/_deploy.py

```diff
@@ -320,6 +320,35 @@ def run(self, deployer):
 
 
 @implementer(IStateChange)
+@attributes(["dataset"])
+class DeleteDataset(object):
+    """
+    Delete all local copies of the dataset.
+
+    A better action would be one that deletes a specific manifestation
+    ("volume" in flocker.volume legacy terminology). Unfortunately
+    currently "remotely owned volumes" (legacy terminology), aka
+    non-primary manifestations or replicas, are not exposed to the
+    deployer, so we have to enumerate them here.
+
+    :ivar Dataset: The dataset to delete.
+    """
+    def run(self, deployer):
+        # service = deployer.volume_service
+        # d = service.enumerate()
+
+        # def got_volumes(volumes):
+        #     deletions = []
+        #     for volume in volumes:
+        #         if volume.name.dataset_id == self.dataset.dataset_id:
+        #             deletions.append(service.pool.destroy(volume))
+        #     return gatherResults(deletions)
```

Comment:
Some error handling/logging here would probably be nice (given the error-handling behavior of `gatherResults`).